src/routes/docs/products/ai/integrations/perplexity/+page.markdoc

```diff
@@ -13,4 +13,310 @@ Integrating Perplexity into your Appwrite project is simple. This tutorial will
+<link rel="stylesheet" href="https://unpkg.com/@appwrite.io/pink-icons" />
+```
+
+This completes the boilerplate for our HTML file, we'll continue to add the rest of the HTML and AlpineJS code in the next steps.
+{% /section %}
+
+{% section #step-5 step=5 title="Serving HTML" %}
+Now that we have our HTML file, we need to serve it from our function. To do this we'll add the following code to our `main.js` file:
+
+```js
+import { getStaticFile, throwIfMissing } from './utils.js';
+
+export default async ({ req, res, error }) => {
+  throwIfMissing(process.env, ['PERPLEXITY_API_KEY']);
+
+  if (req.method === 'GET') {
+    return res.send(getStaticFile('index.html'), 200, {
+      'Content-Type': 'text/html; charset=utf-8',
+    });
+  }
+};
+```
+
+Notice we also added a check to ensure that the `PERPLEXITY_API_KEY` environment variable is set. This is the API key we'll use to authenticate with the Perplexity API.
+{% /section %}
+
+{% section #step-6 step=6 title="Use the Perplexity API" %}
+Now that we have our HTML file being served, we can begin to add the code that will talk to the perplexity API.
+
+First, we're going to import the OpenAI SDK at the top of our `main.js` file:
+
+```js
+import { OpenAI } from 'openai';
+```
+
+Next after we serve the HTML we're going to add code to validate the body of the request and initialise the OpenAI SDK with the Perplexity API key:
+```js
+const model = process.env['PERPLEXITY_MODEL'] || 'mistral-7b-instruct';
+
+try {
+  throwIfMissing(req.body, ['prompt']);
+} catch (err) {
+  return res.json({ ok: false, error: err.message }, 400);
+}
+
+const openai = new OpenAI({
+  apiKey: process.env.PERPLEXITY_API_KEY,
+  baseURL: 'https://api.perplexity.ai',
+});
+```
+This code also allows us to modify what model we use by setting the `PERPLEXITY_MODEL` environment variable.
+
+Finally, we're going to add the code to make the request to the Perplexity API and return the response:
+
+```js
+try {
+  const response = await openai.chat.completions.create({
+    model: model,
+    max_tokens: parseInt(process.env.PERPLEXITY_MAX_TOKENS ?? '512'),
+    messages: [{ role: 'user', content: req.body.prompt }],
+    stream: false,
+  });
+  const completion = response.choices[0].message?.content;
+  return res.json({ ok: true, completion }, 200);
+} catch (err) {
+  error(err);
+  return res.json({ ok: false, error: 'Failed to query model.' }, 500);
+}
+```
+
+This code will send our prompt to the perplexity chat completions API and return the response to the user,
+additionally it'll also catch any errors we could encounter and reports them for easy debugging.
+{% /section %}
+
+{% section #step-7 step=7 title="Making our page interactive" %}
+Now we have a working API that can talk to the Perplexity API, we can begin to add the code to make our page interactive.
+
+First we're going to add a new script tag to our head tag that will handle our HTTP Request:
+
+```html
```

Comment:
Let's remove the boilerplate we added earlier and go here directly. There's no reason to show them how to serve a basic HTML file that does nothing only to replace it with this later.

It's also hard to track all the adding and removing.